### PR TITLE
feat(mempool)!: Add new `Contains` method to `Mempool` interface

### PR DESCRIPTION
**Pull request by hvanz**
Required for #3658 

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: mempool/nop_mempool.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ func (*NopMempool) FlushAppConn() error { return nil }
 // Flush does nothing.
 func (*NopMempool) Flush() {}
 
+// RemoveTxByKey always returns false.
```
Comment (andynog) [inline mempool/nop_mempool.go:66]:
```suggestion
// Contains always returns false.
```

File: mempool/clist_mempool.go [hunk 2]
```diff
@@ -223,6 +218,11 @@ func (mem *CListMempool) Flush() {
 	mem.removeAllTxs()
 }
 
+func (mem *CListMempool) Contains(txKey types.TxKey) bool {
```
Comment (andynog) [inline mempool/clist_mempool.go:221]: In a lot of the methods we have a comment before the method such as 
```
// Safe for concurrent use by multiple goroutines.
```
should we include something for this method ?

File: .changelog/unreleased/breaking-changes/3659-mempool-add-contains-method.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- `[mempool]` Add new `Contains` method to `Mempool` interface.
```
Comment (andynog) [inline .changelog/unreleased/breaking-changes/3659-mempool-add-contains-method.md:1]:
```suggestion
- `[mempool]` Add a new `Contains` method to the `Mempool` interface.
```

**Review by andynog — APPROVED** [review]
@hvanz, lgtm, made some minor comments/suggestions. Thanks !

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**